contracts/InterchainTokenService.sol

```diff
@@ -377,13 +376,13 @@ contract InterchainTokenService is
         string calldata sourceChain,
         string calldata sourceAddress,
         bytes calldata payload
-    ) public payable whenNotPaused {
+    ) public payable override whenNotPaused {
         uint256 messageType = abi.decode(payload, (uint256));
         if (messageType != MESSAGE_TYPE_INTERCHAIN_TRANSFER) {
             revert InvalidExpressMessageType(messageType);
         }
 
-        if (gateway.isCommandExecuted(commandId)) revert AlreadyExecuted();
+        if (gatewayWithToken().isCommandExecuted(commandId)) revert AlreadyExecuted();
```

Comment:
```suggestion
        if (gateway().isCommandExecuted(commandId)) revert AlreadyExecuted();
```
this should still work. only use `gatewayWithToken` when that functionality is needed